dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePlugin.kt

```diff
@@ -158,6 +156,21 @@ class DdAndroidGradlePlugin @Inject constructor(
         }
     }
 
+    private fun resolveMappingFilePath(
+        extensionConfiguration: DdExtensionConfiguration,
+        outputsDir: File,
+        variant: ApplicationVariant
+    ): String {
+        val customPath = extensionConfiguration.mappingFilePath
```

Comment:
You can't read the extension directly here. When this method is called, you're still configuring the project instance (you're in the `apply plugin` part of the project's `build.gradle` script). 
This means that the value of the `mappingFilePath` will always be the default. 

What you need to do is use the `project.afterEvaluate { }` block to wait until the full `build.gradle` script has been read, and the `mappingFilePath` has been set by the customer.

Reply:
this function : `configureVariantForUploadTask` is being called in the `target.afterEvaluate { }` block above in the code so the value will already be there. I also tested it on my end and it works.